mongodbatlas/config.go

```diff
@@ -55,21 +62,41 @@ func (c *Config) NewClient(ctx context.Context) (interface{}, diag.Diagnostics)
 		return nil, diag.FromErr(err)
 	}
 
+	versionedClient, err := newVersionedClient(client)
+	if err != nil {
+		return nil, diag.FromErr(err)
+	}
+
 	clients := &MongoDBClient{
-		Atlas:  atlasClient,
-		Config: c,
+		Atlas:    atlasClient,
+		AtlasSDK: versionedClient,
+		Config:   c,
 	}
 
 	return clients, nil
 }
 
+func newVersionedClient(client *http.Client) (*atlasSDK.APIClient, error) {
+	opts := []atlasSDK.ClientModifier{
+		atlasSDK.UseHTTPClient(client),
```

Comment:
I would strongly recommend to use `useDebug` value in follow-up jira.
UseDebug is essential for us from a support point of view. 

Example: https://github.com/mongodb/atlas-sdk-go/blob/main/examples/aws_cluster/aws.go#L37

In case of community issues, we can recommend that users to set debug flag and sent us tracing information (including internal tracing IDs) that will show us the view on the backend. 

Debug information is printed to stderr

Reply:
Updated (line 82)